frontend/public/components/overview/resource-overview-details.tsx

```diff
@@ -0,0 +1,42 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+
+import { UIActions } from '../../ui/ui-actions';
+import { K8sKind } from '../../module/k8s';
+import {
+  CogAction,
+  ResourceOverviewHeading,
+  SimpleTabNav,
+} from '../utils';
+
+const stateToProps = ({UI}, ownProps): ResourceOverviewDetailsProps => ({
+  ...ownProps,
```

Comment:
Same comment... I don't think this is necessary.

Reply:
Typing with redux `connect` is really complicated. I couldn't find another good example in the app and this was one way I figured out how to get things working without type errors. I'm sure there is a better way. Maybe @alecmerdler has some suggestion on this?

Reply:
Oh, I see the problem. You might be able to use `Partial<ResourceOverviewDetailsProps>`

Reply:
I'd try

```ts
const stateToProps = ({UI}): Partial<ResourceOverviewDetailsProps> => ({
```

If that doesn't work, this should

```ts
const stateToProps = ({UI}) => ({
```

Reply:
The problem with using Partial is that there is no type definitions for the rest of the props that are being passed to this component. I tried defining a separate type for the props that are passed in, then defining types on the `connect` function itself, but could not figure out how to make that work. Everything I tried gave me type errors. I read this article and many others like it, but could not get this approach to work for some reason:

https://medium.com/knerd/typescript-tips-series-proper-typing-of-react-redux-connected-components-eda058b6727d